pkg/client/create_builder.go

```diff
@@ -212,20 +219,75 @@ func (c *Client) fetchLifecycle(ctx context.Context, config pubbldr.LifecycleCon
 	var uri string
 	var err error
 	switch {
+	case buildpack.HasDockerLocator(config.URI):
+		var lifecycleImage imgutil.Image
+		var blob blob.Blob
+		imageName := buildpack.ParsePackageLocator(config.URI)
+		c.logger.Debugf("Downloading lifecycle image: %s", style.Symbol(imageName))
+
+		lifecycleImage, err = c.imageFetcher.Fetch(ctx, imageName, image.FetchOptions{Daemon: false})
```

Comment:
I think, we need to passthrough some key fetch options: `Daemon`, `PullPolicy`, `Platform`, those values should be available from method caller and we need to configure the Fetcher with them